homeassistant/components/niko_home_control/light.py

```diff
@@ -52,36 +57,23 @@ async def async_setup_platform(
 class NikoHomeControlLight(LightEntity):
     """Representation of an Niko Light."""
 
-    _attr_color_mode = ColorMode.ONOFF
-    _attr_supported_color_modes = {ColorMode.ONOFF}
-
     def __init__(self, light, data):
         """Set up the Niko Home Control light platform."""
         self._data = data
         self._light = light
-        self._unique_id = f"light-{light.id}"
-        self._name = light.name
-        self._state = light.is_on
-
-    @property
-    def unique_id(self):
-        """Return unique ID for light."""
-        return self._unique_id
-
-    @property
-    def name(self):
-        """Return the display name of this light."""
-        return self._name
-
-    @property
-    def is_on(self):
-        """Return true if light is on."""
-        return self._state
+        self._attr_unique_id = f"light-{light.id}"
+        self._attr_name = light.name
+        self._attr_is_on = light.is_on
+        self._attr_color_mode = ColorMode.ONOFF
+        self._attr_supported_color_modes = {ColorMode.ONOFF}
+        if light._state["type"] == 2:
+            self._attr_color_mode = ColorMode.BRIGHTNESS
+            self._attr_supported_color_modes.add(ColorMode.BRIGHTNESS)
```

Comment:
When BRIGHTNESS is used, it should be the only supported more.
```suggestion
            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
```